bundles/paikkatietoikkuna/coordinateconversion/components/select.js

```diff
@@ -0,0 +1,168 @@
+Oskari.clazz.define('Oskari.coordinateconversion.component.select',
+    function ( instance ) {
+        var me = this;
+        me.instance = instance;
+        me.selects = [];
+        me.dropdowns = [];
+    }, {
+        getSelect: function () {
+            return this.selects;
+        },
+        getDropdown: function () {
+            return this.dropdowns;
+        },
+        create: function () {
+            var json = this.instance.helper.getOptionsJSON();
+
+            var selections = [];
+            var dropdowns = [];
+            var selects = [];
+            var options = {}
+            jQuery.each( json, function ( key, value ) {
+                var size = Object.keys( value ).length;
+                 jQuery.each( value, function ( key, val ) {
+                    var valObject = {
+                        id : val.id,
+                        title : val.title,
+                        cls: val.cls
+                    };
+                    selections.push( valObject );
+                    if ( key === "0" ) {
+                        options = {
+                            placeholder_text: val.title,
+                            allow_single_deselect : true,
+                            disable_search_threshold: 10,
+                            width: '100%'
+                        };
+                    }
+                     if ( key == size -1 ) {
+                        var select = Oskari.clazz.create('Oskari.userinterface.component.SelectList', "id");
+                        var dropdown = select.create(selections, options);
+                        selections = [];
+                        options = {};
+
+                        dropdown.css( { width:'170px', float:'right' } );
+                        select.adjustChosen();
+                        select.selectFirstValue();
+
+                        selects.push(select);
+                        dropdowns.push(dropdown);
+                     }
+                });
+            });
+                this.dropdowns = dropdowns;
+                this.selects = selects;
+        },
+        /**
+         * @method handleSelectionChanged
+         * which key corresponds to which dropdown in the array:
+         * [0] = geodetic datum,
+         * [1] = coordinate system
+         * [2] = map projection
+         * [3] = geodetic coordinate system
+         * [4] = heigth system
+         */
+        handleSelectionChanged: function ( element ) {
+            var me = this;
+            var values = [];
+            var instances = this.getSelect();
```

Comment:
The naming in here doesn't feel consistent. We get a select, but receives an array of instances as return value? Same on the next row with getDropdown() returns an array instead of a single dropdown.